batch/batch/cloud/gcp/worker/worker_api.py

```diff
@@ -128,3 +138,111 @@ async def close(self):
+        recursive = request.query.get('recursive')
+        if recursive == 'true':
+            return web.json_response(
+                {
+                    'aliases': ['default'],
+                    'email': gsa_email,
+                    'scopes': ['https://www.googleapis.com/auth/cloud-platform'],
+                },
+            )
+        return web.Response(text='aliases\nemail\nidentity\nscopes\ntoken\n')
+
+    async def user_email(self, request: web.Request):
+        return web.Response(text=self._user_credentials(request).email)
+
+    async def user_token(self, request: web.Request):
+        gsa_email = request.match_info['gsa']
+        creds = self._container_credentials(request)[gsa_email]
+        access_token = await creds.access_token_obj()
+        return web.json_response(
+            {
+                'access_token': access_token.token,
+                'expires_in': access_token.expires_in,
+                'token_type': 'Bearer',
+            }
+        )
+
+    @web.middleware
+    async def configure_response(self, request: web.Request, handler):
+        credentials = self._container_credentials(request)
+        gsa = request.match_info.get('gsa', 'default')
+        if gsa not in credentials.keys():
+            raise web.HTTPBadRequest()
+
+        response = await handler(request)
+        response.enable_compression()
+
+        # `gcloud` does not properly respect `charset`, which aiohttp automatically
+        # sets so we have to explicitly erase it
+        # See https://github.com/googleapis/google-auth-library-python/blob/b935298aaf4ea5867b5778bcbfc42408ba4ec02c/google/auth/compute_engine/_metadata.py#L170
+        if 'application/json' in response.headers['Content-Type']:
+            response.headers['Content-Type'] = 'application/json'
+        response.headers['Metadata-Flavor'] = 'Google'
+        response.headers['Server'] = 'Metadata Server for VM'
+        response.headers['X-XSS-Protection'] = '0'
+        response.headers['X-Frame-Options'] = 'SAMEORIGIN'
+        return response
+
+    def create_app(self) -> web.Application:
+        metadata_app = web.Application(
+            client_max_size=HTTP_CLIENT_MAX_SIZE,
+            middlewares=[self.set_request_credentials, self.configure_response],
+        )
+        metadata_app.add_routes(
+            [
+                web.get('/', self.root),
+                web.get('/computeMetadata/v1/project/project-id', self.project_id),
+                web.get('/computeMetadata/v1/project/numeric-project-id', self.numeric_project_id),
+                web.get('/computeMetadata/v1/instance/service-accounts/', self.service_accounts),
+                web.get('/computeMetadata/v1/instance/service-accounts/{gsa}/', self.user_service_account),
+                web.get('/computeMetadata/v1/instance/service-accounts/{gsa}/email', self.user_email),
+                web.get('/computeMetadata/v1/instance/service-accounts/{gsa}/token', self.user_token),
```

Comment:
I'm not sure when it was introduced, but it seems like there are a couple older versions of the API.

https://github.com/GoogleCloudPlatform/k8s-metadata-proxy/blob/master/metadata/metadata.go

```
/0.1/meta-data/service-accounts/.+/identity
/computeMetadata/v1beta1/instance/service-accounts/.+/identity
/computeMetadata/v1/instance/service-accounts/.+/identity
```

Maybe that test is using a particularly old gcloud?